spec/cypress/e2e/annotations_spec.cy.js

```diff
@@ -0,0 +1,69 @@
+import FactoryBot from "../support/factorybot";
+
+function createLectureLessonMedium(context, teacher) {
+  // Lecture
+  FactoryBot.create("lecture_with_sparse_toc", "with_title",
+    { title: "Groundbreaking lecture", teacher_id: teacher.id }).as("lecture");
+
+  // Lesson
+  cy.then(() => {
+    FactoryBot.create("valid_lesson", { lecture_id: context.lecture.id }).as("lesson");
+  });
+
+  // Medium
+  cy.then(() => {
+    FactoryBot.create("lesson_medium", "with_video", "released",
+      "with_lesson_by_id", { lesson_id: context.lesson.id, description: "Soil medium" })
+      .as("medium");
+  });
+}
+
+describe("Annotations visibility", () => {
+  context("when teacher disables annotation sharing with teachers", () => {
+    it("annotations published before that are still visible to the teacher", function () {
+      cy.createUser("generic").as("user");
+      cy.createUserAndLogin("teacher").then(teacher => createLectureLessonMedium(this, teacher));
+
+      cy.then(() => {
+        // Create new annotation
+        FactoryBot.create("annotation", "with_text", "shared_with_teacher",
+          { medium_id: this.medium.id, user_id: this.user.id }).as("annotation");
+
+        // Disable annotation sharing in lecture settings
+        cy.visit(`/lectures/${this.lecture.id}/edit#communication`);
+        cy.getBySelector("annotation-lecture-settings")
+          .find("input[value=0]").should("have.length", 1).click();
```

Comment:
Would it be possible to wait until the tab is actually displayed? If you watch the test in the GUI, at this point it looks like this:
![Screenshot 2024-08-23 120819](https://github.com/user-attachments/assets/f7e6e9ce-f39f-4622-ade7-07dd2f5d52b9)
I think it would be nicer to actually see what is clicked in the GUI (in order to get closer to the look that the users have) and to faster identify what is wrong if the test should fail in the future at some point.

Reply:
good point, I now wait for respective elements to be visible on the page as of 36ff03b8